public/app/plugins/datasource/graphite/MetricTankMetaInspector.tsx

```diff
@@ -25,7 +25,7 @@ export class MetricTankMetaInspector extends PureComponent<Props, State> {
     let totalSeconds = 0;
 
     for (const bucket of buckets) {
-      totalSeconds += kbn.interval_to_seconds(bucket.retention);
+      totalSeconds += bucket.retention ? kbn.interval_to_seconds(bucket.retention) : 0;
```

Comment:
Could rewrite this using reduce (though this is personal preference of course)

```typescript
const totalSeconds = buckets.reduce((acc, bucket) => acc + (bucket.retention ? kbn.interval_to_seconds(bucket.retention) : 0));
```

Reply:
Neat, I like that solution.